### PR TITLE
Create account

### PR DESCRIPTION
**Pull request by aseidens**
@gcarlton 
Added family specific account creation form, modified database for family specific fields in dbpersons table, added approve page for admins to approve families to become active accounts (or reject them and become inactive accounts)

Need the updated SQL file for the dbpersons table in order for this code to work, shared via discord. Approve does not require login at the moment, does require pending family accounts in the table for anything to show up.

### CODE REVIEW

**Comment by gcarlton** [comment]
The checkbox data needs saving, and the group will decide on what to pick for approving families.

File: approve.php [hunk 1]
```diff
@@ -1,8 +1,36 @@
 <?php
-// connect database
+// Start session
+session_start();
+
+// Connect to the database
```
Comment (aseidens) [inline approve.php:7]: I think there might be a built in function that does the database connection

File: approve.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+$username = "fgp";
+$password = "fgp";
+
+$connection = mysqli_connect($hostname, $username, $password, $database);
+
+// Check if the connection was successful
+if (!$connection) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
```
Comment (aseidens) [inline approve.php:19]: We might want to move this function to the dbpersons file to keep the code better organized

File: approve.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    </tr>
+
+<?php
+while ($row = mysqli_fetch_assoc($result)) {
+    ?>
+    <tr>
+        <td style="text-align: center; padding: 10px;"><?php echo $row['last_name']; ?></td>
+        <td style="text-align: center; padding: 10px;">
+    <?php
+
```
Comment (aseidens) [inline approve.php:64]: Need to decide on consistent capitalization for "status" variable, mixed capitalization here, lines 33 and 36 also use capitalized values.

File: createAccountForm.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    '11:59 PM'
+];
+$values = [
+    "00:00", "01:00", "02:00", "03:00", "04:00", "05:00", 
+    "06:00", "07:00", "08:00", "09:00", "10:00", "11:00", 
+    "12:00", "13:00", "14:00", "15:00", "16:00", "17:00", 
+    "18:00", "19:00", "20:00", "21:00", "22:00", "23:00",
+    "23:59"
+];
+
```
Comment (gcarlton) [inline createAccountForm.php:16]: A comment could explain this function to understand it throughout the group.

File: createAccount.php [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+                $lawncare=0;
+            }
+            
+            if(isset($_POST['housecleaning'])){
+                $housecleaning=1;
+            }
+            else{
+                $housecleaning=0;
+            }
+            /*
```
Comment (gcarlton) [inline createAccount.php:251]: Is this to line 271 needed it's commented out.

File: createAccount.php [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+            /*$phone1 = validateAndFilterPhoneNumber($args['econtact-phone']);
+            if (!$econtactPhone) {
+                $errors = true;
+                echo 'bad e-contact phone';
+            }*/
+            
+            $diagnosis=$args['diagnosis'];
+            $permission_to_confirm=$args['permission_to_confirm'];
+           
+            /*
```
Comment (gcarlton) [inline createAccount.php:143]: Line 143 to 169, are all commented out is this needed?

File: approve.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+}
+?>
+
+</table>
+
+<?php
+// Close the database connection
+mysqli_close($connection);
+require_once('universal.inc');
+?>
```
Comment (aseidens) [inline approve.php:91]: good comments overall. Could have comment at the top of the files describing the general purpose and action. Consistent formatting and descriptive variable names.

File: approve.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    // Display status
+    if ($row['status'] == 'Active') {
+        echo "Approved";
+    } elseif ($row['status'] == 'pending') {
+        echo "Pending Approval";
+    }
+    ?>
+        </td>
+        <td style="text-align: center; padding: 10px;">
+            <form method="post">
```
Comment (aseidens) [inline approve.php:74]: Selecting "approve" needs to direct the admin to a page where they can enter values for the leadVolunteer, gift_card_delivery_method, start_date, and location fields.

Comment (gcarlton) [reply approve.php:74]: I have the page where the admin goes I just need to add where the data is inserted where it needs to be.

File: createAccount.php (hunk 6 above)
Comment (aseidens) [reply createAccount.php:143]: This code was for testing different ways to parse the checkbox question. Lines 159 to 166 are the base for the working code in the current iteration. The rest of the commented out code can be deleted.

File: approve.php (hunk 7 above)
Comment (gcarlton) [reply approve.php:91]: I will start adding comments at the top of the file explaining what the file should do

File: createAccount.php (hunk 5 above)
Comment (aseidens) [reply createAccount.php:251]: Lines 251 to 271 were again part of the testing of the checkbox parsing. Lines 271 to 285 were to ensure the code ran without errors. Lines 251 to 285 can all be deleted.

File: approve.php (hunk 1 above)
Comment (gcarlton) [reply approve.php:7]: Sounds good I'll use that to clean the code up.

File: approve.php (hunk 3 above)
Comment (gcarlton) [reply approve.php:64]: Let's do uppercase.

**Pull request merged**